### PR TITLE
Clarify handling of `src` in `ptr::write`

### LINKED ISSUE #39733: `std::ptr::write` docs don't mention that `src` is leaked
**EDIT:** To summarize it a bit better, since it seems like I worded things a bit unclearly:

`src`'s destructor doesn't get run when it's passed by-value to `ptr::write`, but the docs don't really make this clear. It would be better to clarify this, especially given it's an unsafe function.

---

```rust
#[derive(Debug)]
struct Foo(i32);

impl Drop for Foo {
    fn drop(&mut self) {
        println!("{:?} dropped!", self);
    }
}

fn main() {
    let mut dst = Foo(0);
    let src = Foo(1);
    
    unsafe { std::ptr::write(&mut dst, src); }
    
    println!("dst = {:?}", dst);
}
```

This code prints:
```
dst = Foo(1)
Foo(1) dropped!
```
demonstrating that the original value stored in `dst`, and the value passed to `std::ptr::write`, were not dropped; only `dst` was dropped after the `println!` line.

However, [the docs for `std::ptr::write`](https://doc.rust-lang.org/nightly/std/ptr/fn.write.html) only mention that the former isn't dropped, saying nothing about the latter.

### PR DESCRIPTION
**Pull request by tbu-**
Fixes #39733.

### CODE REVIEW

**Comment by tbu-** [comment]
CC  @niconii

**Comment by nagisa** [comment]
The description seems overly contrived to me. I had to reread it multiple times and look at the issue to understand what the paragraph is getting at.

**Comment by tbu-** [comment]
@nagisa Mh. :/

I know this is just a single sentence but can you pinpoint why or what part caused you to reread that sentence multiple times?

**Comment by niconii** [comment]
@tbu- I also think there are a few ways the wording could be improved.

- Saying "it does not drop the contents of `src` *immediately*" seems potentially confusing, since "it" here means "`ptr::write`", making it sound like `ptr::write` itself could drop it at some later point.

- Talking about "the contents of `src`" to parallel "the contents of `dst`" is also not great given `dst` is a pointer whereas `src` is a value passed directly into the function.

- The thing being referred to by "it" changes from "`ptr::write`" to "the contents of `src`" somewhat abruptly.

Personally, I would reword it like this:

>In addition, it does not drop `src`. Semantically, `src` is moved into the location pointed to by `dst`.

**Comment by nagisa** [comment]
What Nicolette said.
